### PR TITLE
Use ST_PointOnSurface for stations

### PR DESCRIPTION
**Pull request by pnorman**
Also a minor code cleanup to prio generation

### CODE REVIEW

File: project.mml [hunk 1]
```diff
@@ -1423,40 +1422,38 @@ Layer:
...
+            station
+          FROM
+          (SELECT
+              ST_PointOnSurface(way) AS way,
+              name,
+              ref,
+              railway,
+              aerialway,
+              tags->'station' AS station
+            FROM planet_osm_polygon
```
Comment (imagico) [inline project.mml]: Don't you need  a `way && !bbox!` here as well?

Comment (imagico) [inline project.mml]: Don't you need a `way && !bbox!` here as well?

**Comment by matthijsmelissen** [comment]
> Note that this adds rendering of railway=subway_entrance on polygons

I'd prefer the ST_PointOnSurface changes not to have impact on rendering output. If there is something in the rendering we want to change, we can do so afterwards in a separate pull request.

**Comment by matthijsmelissen** [comment]
> Note that this adds rendering of railway=subway_entrance on polygons

I'd prefer the ST_PointOnSurface changes not to have impact on rendering output. If there is something in the rendering we want to change, we can do so afterwards in a separate pull request.  That way we can keep the technical discussion separate from the discussion on how we want things to render.

**Comment by pnorman** [comment]
I've fixed it to mostly match the existing rendering. Where it doesn't match is the ordering which was previously undefined. It now also orders by type of station before geometry type, because that's possible now.

**Pull request merged**